runtime/remote_config.go

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package zanzibar
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/pkg/errors"
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+)
+
+// RemoteConfig allow accessing values from a periodically refreshed json file
+type RemoteConfig interface {
+	GetBoolean(string) (bool, error)
+	GetFloat(string, float64) (float64, error)
+	GetInt(string, int64) (int64, error)
+	GetString(string, string) (string, error)
+	GetStruct(string, interface{}) error
```

Comment:
`GetStruct()` will need a default value too `GetStruct(string, interface{}, interface{}) error`

Reply:
makes sense

Reply:
Actually that doesn't make any sense. Maybe `GetStruct(string, interface{}, interface{}) interface{}` but that's a bit annoying.

Probably `GetStruct(string, interface{}) bool` is best then it can be used as

```
var MyData data;
ok := rc.GetStruct('a-key', &data)
if !ok {
  data = defaultData
}
```

Reply:
lol